src/tools/MonarchPeasantSample/main.cpp

```diff
@@ -0,0 +1,175 @@
+﻿#include "pch.h"
+#include <conio.h>
+#include "Monarch.h"
+#include "Peasant.h"
+#include "AppState.h"
+#include "../../types/inc/utils.hpp"
+
+using namespace winrt;
+using namespace winrt::Windows::Foundation;
+using namespace ::Microsoft::Console;
+
+////////////////////////////////////////////////////////////////////////////////
+// This seems like a hack, but it works.
+//
+// This class factory works so that there's only ever one instance of a Monarch
+// per-process. Once the first monarch is created, we'll stash it in g_weak.
+// Future callers who try to instantiate a Monarch will get the one that's
+// already been made.
+//
+// I'm sure there's a better way to do this with WRL, but I'm not familiar
+// enough with WRL to know for sure.
+
+winrt::weak_ref<MonarchPeasantSample::implementation::Monarch> g_weak{ nullptr };
+
+struct MonarchFactory : implements<MonarchFactory, IClassFactory>
+{
+    MonarchFactory() :
+        _guid{ Monarch_clsid } {};
+
+    HRESULT __stdcall CreateInstance(IUnknown* outer, GUID const& iid, void** result) noexcept final
+    {
+        *result = nullptr;
+        if (outer)
+        {
+            return CLASS_E_NOAGGREGATION;
+        }
+
+        if (!g_weak)
+        {
+            // Create a new Monarch instance
+            auto strong = make_self<MonarchPeasantSample::implementation::Monarch>();
+
+            g_weak = (*strong).get_weak();
+            return strong.as(iid, result);
+        }
+        else
+        {
+            // We already instantiated one Monarch, let's just return that one!
+            auto strong = g_weak.get();
+            return strong.as(iid, result);
+        }
+    }
+
+    HRESULT __stdcall LockServer(BOOL) noexcept final
+    {
+        return S_OK;
+    }
+
+private:
+    winrt::guid _guid;
```

Comment:
can I be const since I'm init'd at construction and hopefully never changed?

Reply:
Even better, this guy's not even used so I'll kill it